drift-integration-tests/src/test/java/io/airlift/drift/integration/TestClientsWithApacheServer.java

```diff
@@ -0,0 +1,177 @@
+import org.apache.thrift.server.TServer;
+import org.apache.thrift.server.TServer.Args;
+import org.apache.thrift.server.TSimpleServer;
+import org.apache.thrift.transport.TFramedTransport;
+import org.apache.thrift.transport.TServerSocket;
+import org.apache.thrift.transport.TTransportException;
+import org.apache.thrift.transport.TTransportFactory;
+import org.testng.annotations.Test;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLServerSocket;
+
+import java.util.List;
+import java.util.function.ToIntFunction;
+
+import static com.google.common.collect.Iterables.concat;
+import static com.google.common.collect.Lists.newArrayList;
+import static io.airlift.drift.integration.ApacheThriftTesterUtil.apacheThriftTestClients;
+import static io.airlift.drift.integration.ClientTestUtils.MESSAGES;
+import static io.airlift.drift.integration.DriftNettyTesterUtil.driftNettyTestClients;
+import static io.airlift.drift.integration.LegacyApacheThriftTesterUtil.legacyApacheThriftTestClients;
+import static io.airlift.drift.transport.netty.DriftNettyClientConfig.Transport.HEADER;
+import static java.util.Collections.nCopies;
+import static org.testng.Assert.assertEquals;
+
+public class TestClientsWithApacheServer
+{
+    @Test
+    public static void testApacheServer()
+            throws Exception
+    {
+        testApacheServer(ImmutableList.of());
+    }
+
+    @Test
+    public void testHandlersWithApacheServer()
+            throws Exception
+    {
+        TestFilter firstFilter = new TestFilter();
+        TestFilter secondFilter = new TestFilter();
+        List<MethodInvocationFilter> filters = ImmutableList.of(firstFilter, secondFilter);
+
+        int invocationCount = testApacheServer(filters);
+
+        firstFilter.assertCounts(invocationCount);
+        secondFilter.assertCounts(invocationCount);
+    }
+
+    private static int testApacheServer(List<MethodInvocationFilter> filters)
+            throws Exception
+    {
+        ScribeService scribeService = new ScribeService();
+        TProcessor processor = new scribe.Processor<>(scribeService);
+
+        int invocationCount = 0;
+        for (boolean secure : ImmutableList.of(true, false)) {
+            for (Transport transport : Transport.values()) {
+                for (Protocol protocol : Protocol.values()) {
+                    invocationCount += testApacheServer(secure, transport, protocol, processor, ImmutableList.<ToIntFunction<HostAndPort>>builder()
+                            .addAll(legacyApacheThriftTestClients(filters, transport, protocol, secure))
+                            .addAll(driftNettyTestClients(filters, transport, protocol, secure))
+                            .addAll(apacheThriftTestClients(filters, transport, protocol, secure))
+                            .build());
+                }
+            }
+        }
+
+        assertEquals(scribeService.getMessages(), newArrayList(concat(nCopies(invocationCount, MESSAGES))));
```

Comment:
The `newArrayList(concat(...))` part seems useless. I think it can just be `nCopies`